test/excluded_prefixes_collector_test.go

```diff
@@ -0,0 +1,228 @@
+
+	expectedPrefixes := append(kubeAdmPrefixes, userConfigPrefixes...)
+	et.Eventually(et.checkPrefixes(expectedPrefixes), time.Second*30, time.Second)
+
+	expectedPrefixes = []string{"0.0.0.0/0"}
+	et.Require().NoError(exechelper.Run("kubectl replace -f ./files/updatedUserConfigMap.yaml", et.options...))
+	et.Eventually(et.checkPrefixes(expectedPrefixes), time.Second*15, time.Second)
+
+	et.Require().NoError(exechelper.Run("kubectl delete configmaps excluded-prefixes-config", et.options...))
+	et.Eventually(et.checkPrefixes(kubeAdmPrefixes), time.Second*15, time.Second)
+}
+
+func (et *ExcludedPrefixesSuite) TestWithAllPrefixes() {
+	envPrefixes := []string{
+		"127.0.0.0/8",
+		"134.65.0.0/16",
+	}
+
+	et.Require().NoError(k8s.ApplyDaemonSet("../deployments/prefixes-collector/collector.yaml", func(collector *v1.DaemonSet) {
+		collector.Spec.Template.Spec.Containers[0].Env = append(collector.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  excludedPrefixesEnv,
+			Value: strings.Join(envPrefixes, ","),
+		})
+	}))
+
+	et.Require().NoError(exechelper.Run("kubectl apply -f ./files/userConfigMap.yaml", et.options...))
+	defer func() {
+		et.Require().NoError(exechelper.Run("kubectl delete configmaps excluded-prefixes-config", et.options...))
+	}()
+
+	expectedPrefixes := append(append(kubeAdmPrefixes, envPrefixes...), userConfigPrefixes...)
+	et.Eventually(et.checkPrefixes(expectedPrefixes), time.Second*30, time.Second)
+}
+
+func (et *ExcludedPrefixesSuite) TestWithCorrectEnvPrefixes() {
+	envPrefixes := []string{
+		"127.0.0.0/8",
+		"134.65.0.0/16",
+	}
+
+	et.Require().NoError(k8s.ApplyDaemonSet("../deployments/prefixes-collector/collector.yaml", func(collector *v1.DaemonSet) {
+		collector.Spec.Template.Spec.Containers[0].Env = append(collector.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  excludedPrefixesEnv,
+			Value: strings.Join(envPrefixes, ","),
+		})
+	}))
+
+	expectedPrefixes := append(kubeAdmPrefixes, envPrefixes...)
+	et.Eventually(et.checkPrefixes(expectedPrefixes), time.Second*30, time.Second)
+}
+
+func (et *ExcludedPrefixesSuite) TestWithIncorrectEnvPrefixes() {
+	envPrefixes := []string{
+		"306.306.306.0",
+	}
+
+	et.Require().NoError(k8s.ApplyDaemonSet("../deployments/prefixes-collector/collector.yaml", func(collector *v1.DaemonSet) {
+		collector.Spec.Template.Spec.Containers[0].Env = append(collector.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  excludedPrefixesEnv,
+			Value: strings.Join(envPrefixes, ","),
+		})
+	}))
+
+	et.Eventually(func() bool {
+		podInfo, err := k8s.GetPod(prefixesFileName, "", map[string]string{
+			"app": "excluded-prefixes-collector",
+		})
+		et.Require().NoError(err)
+		return podInfo == nil
+	}, time.Second*15, time.Second)
+}
+
+func TestExcludedPrefixesSuite(t *testing.T) {
+	suite.Run(t, &ExcludedPrefixesSuite{})
+}
+
+func (et *ExcludedPrefixesSuite) checkPrefixes(expectedPrefixes []string) func() bool {
```

Comment:
This func can be converted to inline

```
require.Eventually(func() bool {
  var sb strings.Builder
  err := exechelper.Run('kubectl exec -ti $POD_NAME -- cat $PATH_TO_FILE", exechelper.WithStdOut(&sb))
   return err == nill
})
```

Reply:
Left it as a function, because since we don't wait for alpine pod start, we can't remove code block with getting pod name.

Reply:
Can we split this function into two parts:
1. Find Pod
2. Check configs?
I guess finding pod we can do in SetupTest and then part this checking configs part should look easier.

Reply:
Moved pod finding part to SetupSuite.